pandas/core/arrays/boolean.py

```diff
@@ -653,7 +654,11 @@ def _cmp_method(self, other, op):
             with warnings.catch_warnings():
                 warnings.filterwarnings("ignore", "elementwise", FutureWarning)
                 with np.errstate(all="ignore"):
-                    result = op(self._data, other)
+                    method = getattr(self._data, f"__{op.__name__}__")
+                    result = method(other)
+
+                if result is NotImplemented:
+                    result = invalid_comparison(self._data, other, op)
```

Comment:
This matches what is done in `IntegerArray`.

Reply:
great, in a followon if its possible to move these to the base class would be great.